dg/commands/fyre/cluster/transfer_ownership.py

```diff
@@ -35,17 +35,24 @@
 @click.option("--fyre-api-key", help="FYRE API key (see https://fyre.svl.ibm.com/account)", required=True)
 @click.option("--cluster-name", help="Name of the OCP+ cluster to be transferred", required=True)
 @click.option("--comment", help="Comment")
-@click.option("--new-owner", help="User ID, username, or e-mail address of new owner", required=True)
+@click.option("--new-owner", help="User ID, username, or e-mail address of new owner")
+@click.option("--new-product-group-id", help="ID of new FYRE product group", type=click.INT)
 @click.option("--site", help="OCP+ site", type=click.Choice(["rtp", "svl"]))
+@click.pass_context
 def transfer_ownership(
+    ctx: click.Context,
     fyre_user_name: str,
     fyre_api_key: str,
     cluster_name: str,
     comment: Optional[str],
-    new_owner: str,
+    new_owner: Optional[str],
+    new_product_group: Optional[int],
     site: Optional[str],
 ):
-    """Transfer ownership of an OCP+ cluster"""
+    """Transfer ownership of an OCP+ cluster to a new user and/or product group"""
+
+    if new_owner is None and new_product_group is None:
+        raise click.UsageError("You must set options '--new-owner' and/or '--new-product-group-id'.", ctx)
```

Comment:
what is the reason of having both of these parameters optional without a default value but request for a user input for both of them? (new_owner and new_product_group)

Reply:
Both parameters are optional because you must either specify one of them or both of them. The logic here checks whether none of them was specified, which is not allowed. Unfortunately, Click does not allow specifying relationships between options.

Reply:
In that case why checking on line 54 if both of these parameters are specified? maybe you need `or` instead of `and` in line 54.